moped-editor/src/views/projects/projectsListView/useProjectListViewQuery/useAdvancedSearch.js

```diff
@@ -0,0 +1,83 @@
+import { useState, useMemo } from "react";
+import { useLocation } from "react-router-dom";
+
+const useFilterQuery = (locationSearch) =>
+  useMemo(() => {
+    return new URLSearchParams(locationSearch);
+  }, [locationSearch]);
+
+/**
+ * if filter exists in url, decodes base64 string and returns as object
+ * Used to initialize filter state
+ * @return Object
+ */
+const useMakeFilterState = (filterQuery) =>
+  useMemo(() => {
+    if (Array.from(filterQuery).length > 0) {
+      try {
+        return JSON.parse(atob(filterQuery.get("filter")));
+      } catch {
+        return {};
+      }
+    }
+    return {};
+  }, [filterQuery]);
+
+export const useAdvancedSearch = () => {
+  const filterQuery = useFilterQuery(useLocation().search);
+  const initialFilterState = useMakeFilterState(filterQuery);
+
+  /**
+   * Stores objects storing a random id, column, operator, and value.
+   * @type {Object} filters
+   * @function setFilter - Sets the state of filters
+   * @default {if filter in url, use those params, otherwise {}}
+   */
+  const [filters, setFilters] = useState(initialFilterState);
+
+  const advancedSearchWhereString = useMemo(() => {
```

Comment:
The code in this function is copy/paste from the existing code. Only difference is that it returns a string to place in the graphql query instead of mutating data in the gqlabstract instance.

Reply:
it's very cool to see this in action!

since it has so much logic, you might move this out into a separate function like you did with other pieces of this hook.

Reply:
yes, also think that would be for the best. Making this update right now 🙏